ci/before_install.sh

```diff
@@ -59,3 +59,16 @@ if [[ -n "$CMAKE_URL" ]]; then
     exit 1
   }
 fi
+
+# Install updated ccache.
+(
+  set -ex
+  version=3.7.1
+  cd /tmp
+  curl -L --retry 5 --silent --fail "https://github.com/ccache/ccache/releases/download/v${version}/ccache-${version}.tar.bz2" | tar xj
+  cd "/tmp/ccache-${version}"
+  ./configure
+  make
+  sudo make install
+  ccache --version
+)
```

Comment:
Might be good to test latest ccache in one job (this one) still?
(similar to testing oldest CMake)